desktop-exporter/trace_store.go

```diff
@@ -0,0 +1,73 @@
+package desktopexporter
+
+import (
+	"container/list"
+	"context"
+	"errors"
+	"sync"
+)
+
+const (
+	MAX_QUEUE_SIZE = 10000
+)
+
+type TraceStore struct {
+	mut        sync.Mutex
+	traceQueue *list.List
+	traceMap   map[string][]SpanData
+}
+
+func NewTraceStore() *TraceStore {
+	return &TraceStore{
+		mut:        sync.Mutex{},
+		traceQueue: list.New(),
+		traceMap:   make(map[string][]SpanData),
+	}
+}
+
+func (store *TraceStore) Add(_ context.Context, spanData SpanData) {
+	store.mut.Lock()
+	defer store.mut.Unlock()
+
+	// Enqueue, then append, as the enqueue process checks if the traceID is already in the map to keep the trace alive
+	store.enqueueTrace(spanData.TraceID)
+	store.traceMap[spanData.TraceID] = append(store.traceMap[spanData.TraceID], spanData)
+}
+
+func (store *TraceStore) enqueueTrace(traceID string) {
+	// Make room for the trace in the queue if need be
+	for store.traceQueue.Len() >= MAX_QUEUE_SIZE {
+		store.dequeueTrace()
+	}
+
+	// If the traceID is already in the queue, move it to the back of the line
```

Comment:
This seems backwards to me, but I'm not sure why. Leaving a comment explaining that this is the case is 💯

Reply:
I will switch it the right way around in the next pr.